connector/countconnector/counter.go

```diff
@@ -117,3 +125,24 @@ func (c *counter[K]) appendMetricsTo(metricSlice pmetric.MetricSlice) {
 		}
 	}
 }
+
+func getNestedAttribute(attr string, depth int, attributes pcommon.Map) (pcommon.Value, bool) {
```

Comment:
```suggestion
func getNestedAttribute(attr string, maxDepth int, attributes pcommon.Map) (pcommon.Value, bool) {
```
I think this is slightly more clear.